api/v1alpha1/nutanix_clusterconfig_types.go

```diff
@@ -23,27 +19,16 @@ const (
 type NutanixSpec struct {
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	// host can be either DNS name or ip address
-	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+	ControlPlaneEndpoint ControlPlaneEndpointSpec `json:"controlPlaneEndpoint"`
```

Comment:
I guess this was done to get the `kubebuilder` annotations for the validation?

Reply:
TBH I'm not a fan of using APIs from different modules, would rather copy them over. This reduces dependencies for anyone importing this module. But yeah in this case it was copied to add validations and descriptions via comments.